scripts/entrypoint.sh

```diff
@@ -6,47 +6,48 @@ set -o pipefail
 # and customizes the entrypoint based on what the user has provided.
 # We derive variables from the environment instead of command line
 
-COMMAND="/bin/bash /build_static_site.sh"
-
-# Target is required, project has a default
 if [ -z "${SC_TARGET}" ]; then
    SC_TARGET="${GITHUB_WORKSPACE}/docs"        
 fi
-COMMAND="${COMMAND} --target ${SC_TARGET}"
-COMMAND="${COMMAND} --project ${SC_PROJECT}"
-
-if [ ! -z "${SC_PROJECT_FILE}" ]; then
-    COMMAND="${COMMAND} --project-file ${GITHUB_WORKSPACE}/${SC_PROJECT_FILE}"
-fi
-
-if [ ! -z "${SC_WEIGHTS}" ]; then
-    COMMAND="${COMMAND} --weights ${GITHUB_WORKSPACE}/${SC_WEIGHTS}"
-fi
 
 # Show the user where we are
 echo "Present working directory is:"
-echo "${PWD}"
+pwd
 ls
 
 # Clean up any previous runs (deployed in docs folder)
 # This command needs to be run relative to sourcecred respository
 # that is located at the WORKDIR /code
 rm -rf "${SC_TARGET}"
-echo "${COMMAND}"
-${COMMAND}
+(
+    set -x;
+    /build_static_site.sh \
+        --target "${SC_TARGET}" \
+        --project "${SC_PROJECT}" \
+        ${SC_PROJECT_FILE:+--project-file "${GITHUB_WORKSPACE}/${SC_PROJECT_FILE}"} \
+        ${SC_WEIGHTS:+--weights "${GITHUB_WORKSPACE}/${SC_WEIGHTS}"} \
```

Comment:
I'm not familiar with the syntax here, could you tell me the logic?

Reply:
As mentioned in the commit message, this is called alternate value
expansion. You can get a good explanation of it reading `man bash` and
searching for “alternate value”:

> `${parameter:+word}`: **Use Alternate Value.**  If `parameter` is null
> or unset, nothing is substituted, otherwise the expansion of `word` is
> substituted.

The previous commit message mentions the motivation:

> Fix unquoted parameter expansions
>
> The statements patched in this commit behaved incorrectly when the
> parameters in question contained whitespace or globs.

Reply:
Thank you!